reckoner/chart.py

```diff
@@ -191,63 +185,162 @@ def install(self, namespace=None, context=None):
             self._context = context
 
+        # Fire the pre_install_hook
         self.pre_install_hook()
+
         # TODO: Improve error handling of a repository installation
+        #       Thoughts here, perhaps it would be better to install the
+        #       repositories *before* trying to install the chart. This
+        #       way we could find out earlier our course is wrong.
         self.repository.install(self.name, self.version)
-        self.chart_path = self.repository.chart_path
 
         # Update the helm dependencies
         self.update_dependencies()
 
         # Build the args for the chart installation
         # And add any extra arguments
+        self.build_helm_arguments_for_chart()
 
-        self.args = ['{}'.format(self._release_name), self.chart_path, ]
+        # Check and Error if we're missing required env vars
+        self.__check_env_vars()
+
+        # Perform the upgrade with the arguments
+        try:
+            # Try to run helm upgrade
+            helm_command_response = self.helm.upgrade(self.args)
+            # Log the stdout response in info
+            logging.info(helm_command_response.stdout)
+        except HelmClientException, error:
+            logging.error(error)
+            return
+
+        # Fire the post_install_hook
+        self.post_install_hook()
+
+    def build_helm_arguments_for_chart(self):
+        '''
+        This method builds all the arguments we'll pass along to the helm
+        client once we need to run the install
+        '''
+
+        # Set Default args (release name and chart path)
+        self.args = [
+            '{}'.format(self._release_name),
+            self.repository.chart_path,
+        ]
+
+        # Add namespace to args
         self.args.append('--namespace={}'.format(self.namespace))
+
+        # Add kubecfg context
         if self.context is not None:
             self.args.append('--kube-context={}'.format(self.context))
+
+        # Add debug arguments
         self.args.extend(self.debug_args)
+
+        # Add the version arguments
         if self.version:
             self.args.append('--version={}'.format(self.version))
-        for file in self.files:
-            self.args.append("-f={}".format(file))
-
-        for key, value in self.values.iteritems():
-            for k, v in self._format_set(key, value):
-                self.args.append("--set={}={}".format(k, v))
 
+        # HACK: This is in place until we can fully deprecate the usage of set
+        #       in place of values. Currently values: gets translated into
+        #       --set commands to the helm command line run. This should be
+        #       changed so that set: goes to command line args and values: goes
+        #       to temporary values files, to keep strong types in yaml.
+        #       This will go away once we fully change the functionality of
+        #       values: vs sets: settings in course.yml
+        self._merge_set_and_values()
```

Comment:
This is the main function that enables the migration path between values and sets. the idea is that we set all values as sets and then at some point in the future we transition it.